### PR TITLE
feat: Add active discovery to subnets

### LINKED ISSUE #283: Allow configuring subnet option: Managed allocation and Active discovery
For terraform resource maas_subnet 

It is possible to configure allow dns resolution (allow_dns) and proxy access (allow_proxy) but not managed allocation and active discovery. Allow configuring subnet option: Managed allocation and Active discovery.

### PR DESCRIPTION
**Pull request by skyecas**
## Description of changes

Introduces the `managed` and `active_discovery` fields to subnets, where previously they were `true` and `undefined`.

Both default to true, and can be specified false with:

```terraform
resource "maas_subnet" "test" {
	active_discovery = false
	cidr		 = 10.10.0.0/24
	managed		 = false
	name		 = "test subnet"
}
```

## Issue or ticket link (if applicable)

Resolves: #283 

## Checklist

- [x] I have written a PR title that follows the advice in DEVELOPMENT.md with the title format `type: title`
- [x] I have written a description of the changes in the PR that is clear and concise.
- [x] I have updated the documentation to reflect the changes.
- [x] I have added or updated the tests to reflect the changes.
- [x] I have run the unit tests and they pass.
- [ ] I have run the acceptance tests and they pass.

### CODE REVIEW

**Comment by SK1Y101** [comment]
Requires latest version of GoMAASClient to be included

File: maas/resource_maas_subnet_test.go [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+	allow_dns		 = %t
+	allow_proxy		 = %t
+	cidr			 = %q
+	dns_servers		 = ["8.8.8.8"]
+	gateway_ip		 = %q
+	managed			 = %t
+	name			 = %q
+	fabric			 = "0"
+	vlan			 = "0"
+}`, activeDiscovery, allowDNS, allowProxy, cidr, gatewayIP, managed, name)
```
Comment (tobiasdemendonca) [inline maas/resource_maas_subnet_test.go:110]: Nit: turn tabs -> spaces for config indent

File: maas/resource_maas_subnet.go [hunk 2]
```diff
@@ -85,9 +93,10 @@ func resourceMAASSubnet() *schema.Resource {
 				Description:      "Gateway IP address for the new subnet. This argument is computed if it's not set.",
 			},
 			"ip_ranges": {
-				Type:        schema.TypeSet,
-				Optional:    true,
-				Description: "A set of IP ranges configured on the new subnet. Parameters defined below. This argument is processed in [attribute-as-blocks mode](https://www.terraform.io/docs/configuration/attr-as-blocks.html).",
+				Type:          schema.TypeSet,
+				Optional:      true,
+				ConflictsWith: []string{"managed"},
```
[the remark this answers is not in the document]
Comment (tobiasdemendonca) [reply maas/resource_maas_subnet.go:98]: I am in the process of checking what the implications of this are but you might be faster!

- Not sure if the changing managed to unmanaged definitely changes this behaviour on the fly, and all ip range meanings invert.
- I assume that having an unmanged subnet on a vlan with our new dhcp resource enabled is fine but again need to check this.

**Review by tobiasdemendonca — CHANGES_REQUESTED** [review]
I think this does need a change but I'm being slow to test out these scenarios so just wanted to flag this early incase you also have time to think about the edge cases
